Add nf-core-tools, and three remainder deps

marcodelapierre opened the pull request:
Key changes:
* Added nf-core-tools recipe, and its large python dependency tree (https://nf-co.re/tools/ ; https://pypi.org/project/nf-core/)
* Renamed tower-agent and tower-cli to nf-tower-agent and nf-tower-cli
* Added new package versions for nf-tower-cli and nextflow

Note I have tested the target new tool, `nf-core-tools`, for some functionalities, against a standard pip installation.

marcodelapierre committed:
nextflow recipe: added latest stable version

marcodelapierre committed:
tower-cli recipe: added latest release

marcodelapierre committed:
recipes tower-agent and tower-cli renamed to nf-tower-agent and nf-tower-cli

marcodelapierre committed:
recipes nf-tower-agent and nf-tower-cli: small fix

marcodelapierre committed:
nf-core-tools recipe: added most py- dependencies

marcodelapierre committed:
nf-core-tools: recipe without galaxy-tool-util (for testing)

marcodelapierre committed:
fixed typos in py-yacman recipe

marcodelapierre committed:
fixed typos in py-pytest-workflow recipe

marcodelapierre committed:
fixed typo in nf-core-tools recipe

marcodelapierre committed:
fixed typos in py-yacman recipe

marcodelapierre committed:
fixes in recipes for py-questionary and py-url-normalize

marcodelapierre committed:
Merge branch 'develop' into update/nextflow-tools

marcodelapierre committed:
fixes to py-yacman recipe

marcodelapierre committed:
style fixes to py- packages that are dependencies to nf-core-tools

marcodelapierre committed:
fix in py-requests-cache recipe

marcodelapierre committed:
added missing dep in py-requests-cache recipe

marcodelapierre committed:
Merge branch 'develop' into update/nextflow-tools

marcodelapierre committed:
Merge branch 'develop' into update/nextflow-tools

marcodelapierre committed:
nf-core-tools deps: removed redundant python dep for py packages oyaml and piper

marcodelapierre committed:
nf-core-tools recipe: final, incl dep on py-galaxy-tool-util

marcodelapierre committed:
Merge branch 'develop' of github.com:spack/spack into update/nextflow-tools

marcodelapierre committed:
nf-core-tools: new version with extra dependency

marcodelapierre committed:
added py-galaxy-util, draft: added some required dep versions, still have to add 40+ deps

marcodelapierre committed:
nextflow and nf-core-tools packages: added my self as maintainer

marcodelapierre committed:
style fixes

marcodelapierre committed:
Merge branch 'develop' into update/nextflow-tools

marcodelapierre committed:
style fix for nf-core-tools recipe

marcodelapierre committed:
added license to py-logmuse recipe

marcodelapierre committed:
audit fixes

marcodelapierre committed:
style fix after audit fix

marcodelapierre committed:
py-galaxy-tool-util: added deps 1st bunch

marcodelapierre committed:
audit/style fixes, including adding missing dep package

marcodelapierre committed:
more audit/style fixes

marcodelapierre committed:
more more audit/style fixes

marcodelapierre committed:
moooore audit fixes

marcodelapierre committed:
py-galaxy-tool-util: dependencies 2nd chunk

marcodelapierre committed:
silly audit fix

marcodelapierre committed:
py-galaxy-util deps: 3rd bunch - first 20 done

marcodelapierre committed:
fixes

marcodelapierre committed:
style fix

marcodelapierre committed:
py-galaxy-tool-util: 4th bunch of deps

marcodelapierre committed:
stashing dep recipe backbones for py-galaxy-tool-util

marcodelapierre committed:
Merge branch 'develop' into update/nextflow-tools

marcodelapierre committed:
nf-core-tools: using pre-built wheel for dependency py-galaxy-tool-util

marcodelapierre committed:
nf-core-tools: adding also py-galaxy-util, as wheel

marcodelapierre committed:
fix

marcodelapierre commented:
Note I have also tested the target new tool, `nf-core-tools`, for some functionalities, against a standard pip installation.

pditommaso commented:
Great, thanks!

dialvarezs commented:
Hey @marcodelapierre!
I'm thinking you should add Nextflow 20.10.4 instead of 20.10.3 (or maybe both?).

adamjstewart commented on var/spack/repos/builtin/packages/nf-core-tools/package.py:
```diff
@@ -0,0 +1,45 @@
+from spack.package import *
+
+
+class NfCoreTools(PythonPackage):
```
Is this primarily used as a command line tool or is it primarily imported in a Python script? Any reason to change from the default name `py-nf-core`?

marcodelapierre replied:
hi @adamjstewart , it is one of those cases where the tool "happens" to be written in Python, but is used entirely as a command line tool

adamjstewart replied:
Should we instead call it `nf-core` then?

marcodelapierre replied:
I recommend `nf-core-tools`, because the `nf-core` projects has many repos/sub-projects, with `tools` being the CLI interface

adamjstewart commented on var/spack/repos/builtin/packages/nf-core-tools/package.py:
```diff
@@ -0,0 +1,45 @@
+
+    depends_on("nextflow@21.10.3:", when="@2.2:", type="run")
+
+    depends_on("python@3.7:", when="@2.5:", type=("build", "run"))
```
```suggestion
    depends_on("python@3.7:3", type=("build", "run"))
```

adamjstewart commented on var/spack/repos/builtin/packages/nf-core-tools/package.py:
```diff
@@ -0,0 +1,45 @@
+    version("2.6", sha256="47d4df906a60006249284bbf0bb84cdec48303a699c7c0d0a26f404a50e0811a")
+    version("2.5.1", sha256="0303f6e3810ba1bc6ac843566ee9dea6b5edbf3527437dec5854b8c437456a4f")
+
+    depends_on("nextflow@21.10.3:", when="@2.2:", type="run")
```
```suggestion
    depends_on("nextflow@21.10.3:", type="run")
```
No need to specify if it's required for all versions in the package

adamjstewart commented on var/spack/repos/builtin/packages/nf-tower-agent/package.py:
```diff
@@ -8,7 +8,7 @@
 
 
-class TowerAgent(Package):
+class NfTowerAgent(Package):
```
I usually deprecate the old package when renaming instead of completely deleting it: https://spack.readthedocs.io/en/latest/packaging_guide.html#deprecating-old-versions

marcodelapierre replied:
Very keen on your input on this one, Adam.

I had added `tower-agent` and `tower-cli` earlier this year. Then, while adding `nf-core-tools` with this PR, I have thought it would be friendlier for users to have the first two renamed. (I also believe so far the uptake will have still been low, in terms of spack installations for the former two).
So, according  to the doc page above, it is not even reason number 6 .. the tools themselves have not changed name, it is just a name change for the Spack recipe. And by the way, apologies for this change on the go, I will make sure it will not happen with future Spack contributions from me.

What are your thoughts?

adamjstewart replied:
Number 6 "package rename" refers to the name of the Spack recipe. Or at least, that was my intention when I wrote those docs.

Since you're maintaining these recipes I will likely let you make the final decision. We do frown on package/version rename/removal without first deprecating it and waiting a full release (you can obviously create the new package at the same time as you deprecate the old package), but I don't think we have a hard policy about this.

marcodelapierre replied:
ok, thanks for clarifying Adam.
I have added back the old named packages, and added the `deprecated` property to all their versions

adamjstewart commented on var/spack/repos/builtin/packages/py-attmap/package.py:
```diff
@@ -0,0 +1,21 @@
+
+    version("0.13.2", sha256="fdffa45f8671c13428eb8c3a1702bfdd1123badb99f7af14d72ad53cc7e770de")
+
+    depends_on("python@3.5:", type=("build", "run"))
```
Where do you see this? Honestly I would just remove it, Spack no longer supports Python 3.4 and older anyway

marcodelapierre replied:
done!

adamjstewart commented on var/spack/repos/builtin/packages/py-cattrs/package.py:
```diff
@@ -0,0 +1,24 @@
+    version("22.2.0", sha256="f0eed5642399423cf656e7b66ce92cdc5b963ecafd041d1b24d136fdde7acf6d")
+
+    depends_on("python@3.7:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
```
setuptools isn't needed

marcodelapierre replied:
done!

adamjstewart commented on var/spack/repos/builtin/packages/py-cattrs/package.py:
```diff
@@ -0,0 +1,24 @@
+    depends_on("py-poetry-core@1.1.0:", type="build")
+
+    depends_on("py-attrs@20:", type=("build", "run"))
+    depends_on("py-typing-extensions", when="^python@:3.7.99", type=("build", "run"))
+    depends_on("py-exceptiongroup", when="^python@:3.10.99", type=("build", "run"))
```
```suggestion
    depends_on("py-typing-extensions", when="^python@:3.7", type=("build", "run"))
    depends_on("py-exceptiongroup", when="^python@:3.10", type=("build", "run"))
```

marcodelapierre replied:
done!

adamjstewart commented on var/spack/repos/builtin/packages/py-exceptiongroup/package.py:
```diff
@@ -0,0 +1,20 @@
+    version("1.0.4", sha256="bd14967b79cd9bdb54d97323216f8fdf533e278df937aa2a90089e7d6e06e5ec")
+
+    depends_on("python@3.7:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
```
I don't think setuptools is needed

marcodelapierre replied:
done

adamjstewart commented on var/spack/repos/builtin/packages/py-flit-scm/package.py:
```diff
@@ -0,0 +1,26 @@
+    version("1.7.0", sha256="961bd6fb24f31bba75333c234145fff88e6de0a90fc0f7e5e7c79deca69f6bb2")
+
+    depends_on("python@3.6:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
```
I don't think setuptools is needed

marcodelapierre replied:
done

adamjstewart commented on var/spack/repos/builtin/packages/py-flit-scm/package.py:
```diff
@@ -0,0 +1,26 @@
+    depends_on("python@3.6:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
+
+    depends_on("py-flit-core@3.5:3.5.99", type=("build", "run"))
```
```suggestion
    depends_on("py-flit-core@3.5:3", type=("build", "run"))
```

marcodelapierre replied:
done

adamjstewart commented on var/spack/repos/builtin/packages/py-flit-scm/package.py:
```diff
@@ -0,0 +1,26 @@
+
+    depends_on("py-flit-core@3.5:3.5.99", type=("build", "run"))
+    depends_on("py-setuptools-scm@6.4:", type=("build", "run"))
+    depends_on("py-tomli", type=("build", "run"))
```
```suggestion
    depends_on("py-tomli", when="^python@:3.10", type=("build", "run"))
```

marcodelapierre replied:
done

adamjstewart commented on var/spack/repos/builtin/packages/py-galaxy-tool-util/package.py:
```diff
@@ -0,0 +1,25 @@
+
+    homepage = "https://github.com/galaxyproject/galaxy"
+
+    version(
+        "22.1.5",
+        url="https://files.pythonhosted.org/packages/76/b0/4745ccb75fdb5c48a27a73900f8430119afe6c7c95b1aafed004f093788c/galaxy_tool_util-22.1.5-py2.py3-none-any.whl",
+        sha256="062a758024cda103d3d40ba232379c0c1853e8e181b58a65c4e954456cdcc040",
+        expand=False,
+    )
```
Can you download the source tarball instead?

marcodelapierre replied:
done!

adamjstewart commented on var/spack/repos/builtin/packages/py-galaxy-tool-util/package.py:
```diff
@@ -0,0 +1,25 @@
+    depends_on("python@3.7:3.10", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
+
+    depends_on("py-galaxy-util", type=("build", "run"))
```
Missing many deps:
```
galaxy-util>=22.1
galaxy-containers>=22.1
lxml
pydantic
pyyaml
sortedcontainers
typing-extensions
```

marcodelapierre replied:
all added!

adamjstewart commented on var/spack/repos/builtin/packages/py-galaxy-tool-util/package.py:
```diff
@@ -0,0 +1,25 @@
+        expand=False,
+    )
+
+    depends_on("python@3.7:3.10", type=("build", "run"))
```
Where do you see this?

marcodelapierre replied:
here:   https://github.com/galaxyproject/galaxy/blob/dev/pyproject.toml#L87

adamjstewart replied:
That's for dev, not for 22.1.5. 22.1.5 does not restrict the versions of Python that are supported, so I would remove this line unless you know if doesn't work for 3.11.

marcodelapierre replied:
done

adamjstewart commented on var/spack/repos/builtin/packages/py-galaxy-util/package.py:
```diff
@@ -0,0 +1,25 @@
+
+    homepage = "https://github.com/galaxyproject/galaxy"
+
+    version(
+        "22.1.2",
+        url="https://files.pythonhosted.org/packages/c6/6b/ca93c7a73a1d9c2153592007fd8264a357cb277de8d980ab6aa91c9cc8d1/galaxy_util-22.1.2-py2.py3-none-any.whl",
+        sha256="23c9ea814244dfb020e30ea3284d6dacfd9ba4119fb76c1cc2b3ab379463f43c",
+        expand=False,
+    )
```
Source tarball

marcodelapierre replied:
done!

adamjstewart commented on var/spack/repos/builtin/packages/py-galaxy-util/package.py:
```diff
@@ -0,0 +1,25 @@
+        expand=False,
+    )
+
+    depends_on("python@3.7:3.10", type=("build", "run"))
```
Where?

marcodelapierre replied:
here:    https://github.com/galaxyproject/galaxy/blob/dev/pyproject.toml#L87

adamjstewart commented on var/spack/repos/builtin/packages/py-galaxy-util/package.py:
```diff
@@ -0,0 +1,25 @@
+    depends_on("python@3.7:3.10", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
+
+    depends_on("py-boltons", type=("build", "run"))
```
Missing deps:
```
bleach
boltons
docutils
importlib_resources
MarkupSafe
packaging<22  # packaging 22.0 dropped LegacyVersion
pycryptodome
pyparsing
PyYAML
requests
Routes
typing-extensions
zipstream-new
```

marcodelapierre replied:
all added!

adamjstewart commented on var/spack/repos/builtin/packages/py-logmuse/package.py:
```diff
@@ -0,0 +1,18 @@
+
+    version("0.2.7", sha256="a4692c44ddfa912c3cb149ca4c7545f80119aa7485868fd1412e7c647e9a7e7e")
+
+    depends_on("python@3.5:", type=("build", "run"))
```
Where?

marcodelapierre replied:
they declare it in `setup.py`.
will remove anyway, as you say Spack does not support `@:3.4` any more

adamjstewart commented on var/spack/repos/builtin/packages/py-paramiko/package.py:
```diff
@@ -23,6 +25,7 @@ class PyParamiko(PythonPackage):
     depends_on("py-cryptography@2.5:", when="@2.7:", type=("build", "run"))
     depends_on("py-pyasn1@0.1.7:", when="@:2.1", type=("build", "run"))
     depends_on("py-pynacl@1.0.1:", when="@2.7:", type=("build", "run"))
+    depends_on("py-six", type=("build", "run"))
```
Not needed for all versions

marcodelapierre replied:
where have you found that info?

I have: https://github.com/paramiko/paramiko/blob/main/paramiko/pkey.py#L30

thanks in advance,

adamjstewart replied:
I'm looking at the dependencies listed in `setup.py` under `install_requires` and comparing it across release versions. The oldest version of this package does not list this dep, although the newest does. Haven't checked the in between versions yet.

marcodelapierre replied:
thanks Adam, found, the first version requiring `six` is `2.9.3`

adamjstewart commented on var/spack/repos/builtin/packages/py-pytest-workflow/package.py:
```diff
@@ -0,0 +1,26 @@
+    depends_on("py-setuptools@51:", type="build")
+
+    depends_on("py-pyyaml", type=("build", "run"))
+    depends_on("py-pytest@7.0.0:", type=("build", "run"))
```
```suggestion
    depends_on("py-pytest@5.4:", type=("build", "run"))
```

marcodelapierre replied:
where have you found this info?
this is what  I have: https://github.com/LUMC/pytest-workflow/blob/develop/setup.py#L55

adamjstewart replied:
You're looking at the wrong version of this file: https://github.com/LUMC/pytest-workflow/blob/v1.6.0/setup.py#L55

marcodelapierre replied:
thanks Adam, done

adamjstewart commented on var/spack/repos/builtin/packages/py-questionary/package.py:
```diff
@@ -17,4 +18,5 @@ class PyQuestionary(PythonPackage):
 
     depends_on("python@3.6:3.9", type=("build", "run"))
     depends_on("py-setuptools", type="build")
+    depends_on("py-poetry", type="build")
```
```suggestion
    depends_on("py-poetry@1.0.5:", type="build")
```

marcodelapierre replied:
done

adamjstewart commented on var/spack/repos/builtin/packages/py-questionary/package.py:
```diff
@@ -17,4 +18,5 @@ class PyQuestionary(PythonPackage):
 
     depends_on("python@3.6:3.9", type=("build", "run"))
     depends_on("py-setuptools", type="build")
```
setuptools prob isn't needed anymore

marcodelapierre replied:
checked and done

adamjstewart commented on var/spack/repos/builtin/packages/py-requests-cache/package.py:
```diff
@@ -0,0 +1,30 @@
+    version("0.9.7", sha256="b7c26ea98143bac7058fad6e773d56c3442eabc0da9ea7480af5edfc134ff515")
+
+    depends_on("python@3.7:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
```
setuptools not needed

marcodelapierre replied:
done

adamjstewart commented on var/spack/repos/builtin/packages/py-requests-cache/package.py:
```diff
@@ -0,0 +1,30 @@
+
+    version("0.9.7", sha256="b7c26ea98143bac7058fad6e773d56c3442eabc0da9ea7480af5edfc134ff515")
+
+    depends_on("python@3.7:", type=("build", "run"))
```
```suggestion
    depends_on("python@3.7:3", type=("build", "run"))
```

marcodelapierre replied:
done

adamjstewart commented on var/spack/repos/builtin/packages/py-requests-cache/package.py:
```diff
@@ -0,0 +1,30 @@
+    depends_on("py-cattrs@22.2:", type=("build", "run"))
+    depends_on("py-platformdirs@2.5:", type=("build", "run"))
+    depends_on("py-url-normalize@1.4:", type=("build", "run"))
+    depends_on("py-appdirs", type=("build", "run"))
```
```suggestion
    depends_on("py-appdirs@1.4.4:", type=("build", "run"))
```

marcodelapierre replied:
thanks for spotting!

it is actually `@1.4.4`  :   https://github.com/requests-cache/requests-cache/blob/main/poetry.lock#L542

done

adamjstewart replied:
Lock files aren't used during installation, only the dependencies listed in `pyproject.toml`

marcodelapierre replied:
oh! thanks Adam, did not know that

adamjstewart commented on var/spack/repos/builtin/packages/py-requests-cache/package.py:
```diff
@@ -0,0 +1,30 @@
+    depends_on("py-urllib3@1.25.5:", type=("build", "run"))
+    depends_on("py-attrs@21.2:", type=("build", "run"))
+    depends_on("py-cattrs@22.2:", type=("build", "run"))
+    depends_on("py-platformdirs@2.5:", type=("build", "run"))
```
I don't see platformdirs listed as a dep

marcodelapierre replied:
here it is :   https://github.com/requests-cache/requests-cache/blob/main/pyproject.toml#L39

adamjstewart replied:
You're looking at the wrong version, the most recent stable release does not have this dependency

marcodelapierre replied:
thanks Adam, done

adamjstewart commented on var/spack/repos/builtin/packages/py-ubiquerg/package.py:
```diff
@@ -0,0 +1,19 @@
+
+    version("0.6.2", sha256="a9b1388799d4c366f956e0c912819099ad8f6cd0e5d890923cdde197f80d14cf")
+
+    depends_on("python@3.5:", type=("build", "run"))
```
Where do you see this? would remove

marcodelapierre replied:
done!

adamjstewart commented on var/spack/repos/builtin/packages/py-url-normalize/package.py:
```diff
@@ -0,0 +1,22 @@
+
+    depends_on("python@3.6:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
+    depends_on("py-poetry@1.0.0:", type="build")
```
```suggestion
    depends_on("py-poetry@0.12:", type="build")
```

marcodelapierre replied:
where do you see this?
I have:  https://github.com/niksite/url-normalize/blob/master/pyproject.toml#L33

adamjstewart replied:
https://github.com/niksite/url-normalize/blob/1.4.3/pyproject.toml#L34

marcodelapierre replied:
thanks Adam, done

adamjstewart commented on var/spack/repos/builtin/packages/py-url-normalize/package.py:
```diff
@@ -0,0 +1,22 @@
+    version("1.4.3", sha256="d23d3a070ac52a67b83a1c59a0e68f8608d1cd538783b401bc9de2c0fac999b2")
+
+    depends_on("python@3.6:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
```
setuptools not needed

marcodelapierre replied:
done

adamjstewart commented on var/spack/repos/builtin/packages/py-url-normalize/package.py:
```diff
@@ -0,0 +1,22 @@
+
+    version("1.4.3", sha256="d23d3a070ac52a67b83a1c59a0e68f8608d1cd538783b401bc9de2c0fac999b2")
+
+    depends_on("python@3.6:", type=("build", "run"))
```
```suggestion
    depends_on("python@2.7:2,3.6:3", type=("build", "run"))
```

marcodelapierre replied:
I would leave 2.x out, as it is deprecated by spack, right?

other one done

adamjstewart replied:
You could honestly just say `@:3`, but yes, whatever is fine

marcodelapierre committed:
nextflow: added latest bugfix version

marcodelapierre commented:
> Hey @marcodelapierre!
> I'm thinking you should add Nextflow 20.10.4 instead of 20.10.3 (or maybe both?).

done!

marcodelapierre committed:
Update var/spack/repos/builtin/packages/nf-core-tools/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

marcodelapierre committed:
Update var/spack/repos/builtin/packages/nf-core-tools/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

marcodelapierre committed:
Merge branch 'update/nextflow-tools' of github.com:marcodelapierre/spack into update/nextflow-tools

marcodelapierre committed:
nf-core-tools pr: 1st bunch of review edits

marcodelapierre committed:
nf-core-tools: 2nd bunch of review edits

marcodelapierre committed:
adding back tower-agent and tower-cli as deprecated

marcodelapierre committed:
nf-core-tools: 3rd bunch of review edits

marcodelapierre committed:
small style fix

marcodelapierre committed:
merged from develop

marcodelapierre committed:
Merge branch 'develop' into update/source-nextflow-tools

marcodelapierre commented:
@adamjstewart thanks for your feedback so far, I have much appreciated it.

Now I am going to tackle the issues with `py-galaxy-tool-util` and `py-galaxy-util`.

marcodelapierre committed:
merged from my source branch

marcodelapierre committed:
prepping py-galaxy-tool-util for further work

adamjstewart commented:
Let me know when things are ready for a second round of reviewing!

marcodelapierre committed:
nf-core-tools: last bunch of deps, except for galaxy-tool-util and pulsar

marcodelapierre committed:
audit fixes

marcodelapierre committed:
updates to py-galaxy-tool-util and its deps, still 2 to work on

marcodelapierre committed:
one style fix

marcodelapierre committed:
updated recipe for py-galaxy-util

marcodelapierre committed:
updated recipe for py-pulsar-galaxy-lib

marcodelapierre committed:
Merge branch 'develop' into update/nextflow-tools

adamjstewart commented:
Sqlalchemy was updated in another PR, you'll need to rebase or remove changes to that package.

Also, I would be eternally grateful if you could split this PR into multiple smaller PRs. It's extremely difficult to review PRs once they involve 10+ packages. I realize that some of these packages require modifications to other packages, but if there are any that are independent or a subset that could be merged first, that would speed up the review process.

marcodelapierre committed:
typo fix

marcodelapierre committed:
shasum fixes

marcodelapierre committed:
updated py-sqlalchemy from develop

marcodelapierre committed:
update py-sqlalchemy to merge develop

marcodelapierre commented:
> Sqlalchemy was updated in another PR, you'll need to rebase or remove changes to that package.
> 
> Also, I would be eternally grateful if you could split this PR into multiple smaller PRs. It's extremely difficult to review PRs once they involve 10+ packages. I realize that some of these packages require modifications to other packages, but if there are any that are independent or a subset that could be merged first, that would speed up the review process.

develop merge down!

I will now work to extract a few smaller PRs out of this one. Will ping you inside them.

Unfortunately, the bulk of the new recipes will all stay in the main PR: my goal was to add support for `nf-core-tools` within Spack, and this package comes with a huge dependency tree.

marcodelapierre commented:
All right, currently 89 files are being edited/added in this PR. Let's try and chunk these a bit.

marcodelapierre commented:
First PR shot (5 files): #34515

marcodelapierre commented:
Second PR shot (8  files): #34516

marcodelapierre commented:
@adamjstewart I suggest we get back to this one when the 2 PRs above are merged (hopefully it won't take long, they are both small)

marcodelapierre committed:
added newest versions (today) for nf-tower-agent and nf-tower-cli

marcodelapierre committed:
Merge branch 'develop' into update/nextflow-tools

marcodelapierre commented:
Thanks @adamjstewart ! 

Now going to tackle the big group of 76 new packages.

As a note, as of now I have addressed all suggestions you had made further up in this PR.

marcodelapierre commented:
Now trying to see how I can split this PR.

As I said, it is not trivial, because my goal was to add support for `nf-core-tools`, which has a huge dependency tree.
I just went ahead by brute force and added them all.

However, three stand out as they have in turn lots of dependencies, so I tackled them last: `py-galaxy-tool-util`, `py-galaxy-util` and `py-pulsar-galaxy-lib`.

I am going to try and make at least 2 distinct PRs by keeping this in mind.

marcodelapierre commented:
Third PR shot (16 files): #34537

marcodelapierre committed:
Merge branch 'develop' into update/nextflow-tools

marcodelapierre committed:
Update var/spack/repos/builtin/packages/py-requests-cache/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

marcodelapierre committed:
Update var/spack/repos/builtin/packages/py-requests-cache/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

marcodelapierre committed:
Merge branch 'develop' into update/nextflow-tools

marcodelapierre committed:
Merge branch 'develop' into update/nextflow-tools

marcodelapierre committed:
Merge branch 'update/nextflow-tools' of github.com:marcodelapierre/spack into update/nextflow-tools

marcodelapierre committed:
adding 2nd bunch of nf-core deps from update/nextflow-tools

marcodelapierre committed:
Merge branch 'develop' into update/nextflow-tools

marcodelapierre committed:
Merge branch 'develop' into add/2nd-nf-core-deps

marcodelapierre committed:
adding 3rd bunch of nf-core deps from update/nextflow-tools

marcodelapierre committed:
Merge branch 'add/2nd-nf-core-deps' into add/3rd-nf-core-deps

marcodelapierre committed:
4th chunk of nf-core deps from update/nextflow-tools

marcodelapierre commented:
Sixth PR shot (19 files): #34564

marcodelapierre committed:
Merge branch 'add/3rd-nf-core-deps' into add/4th-nf-core-deps

marcodelapierre commented:
OK @adamjstewart , ofter all of the above PRs are done, 

the current one will only have 4 packages left, and can go next (and last):
```
nf-core-tools
py-galaxy-tool-util
py-galaxy-util
py-pulsar-galaxy-lib
```

marcodelapierre commented:
This is the recommended  list of reviews:

#34537 

#34562 

#34563

#34564

current one: #34466

marcodelapierre commented:
I will be without computer from this afternoon and for about a week (West Australia time).

Feel free to go ahead as time permits; I will come back here when I am back.

Thanks again @adamjstewart with your ongoing support for this (overgrown) contribution effort!

marcodelapierre committed:
merged develop after 1st,2nd round of nf-core deps

marcodelapierre committed:
Merge branch 'add/3rd-nf-core-deps' into add/4th-nf-core-deps

marcodelapierre committed:
merged from add/4th

marcodelapierre committed:
Update var/spack/repos/builtin/packages/py-dnspython/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

marcodelapierre committed:
Update var/spack/repos/builtin/packages/py-dnspython/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

marcodelapierre committed:
Update var/spack/repos/builtin/packages/py-dnspython/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

marcodelapierre committed:
Update var/spack/repos/builtin/packages/py-fastapi-utils/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

marcodelapierre committed:
Update var/spack/repos/builtin/packages/py-pastedeploy/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

marcodelapierre committed:
Update var/spack/repos/builtin/packages/py-pebble/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

marcodelapierre committed:
Update var/spack/repos/builtin/packages/py-fastapi/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

marcodelapierre committed:
Update var/spack/repos/builtin/packages/py-fastapi/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

marcodelapierre committed:
Update var/spack/repos/builtin/packages/py-gunicorn/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

marcodelapierre committed:
Update var/spack/repos/builtin/packages/py-starlette/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

marcodelapierre committed:
Update var/spack/repos/builtin/packages/py-starlette/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

marcodelapierre committed:
Update var/spack/repos/builtin/packages/py-starlette/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

marcodelapierre committed:
Update var/spack/repos/builtin/packages/py-parsley/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

marcodelapierre committed:
Update var/spack/repos/builtin/packages/py-paste/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

marcodelapierre committed:
Update var/spack/repos/builtin/packages/py-paste/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

marcodelapierre committed:
py-gxformat2: added comment

marcodelapierre committed:
py-lagom: now using github tarballs

marcodelapierre committed:
fix for py-lagom

marcodelapierre committed:
adding missing deps to py-fastapi-utils

marcodelapierre committed:
another fix to py-lagom

marcodelapierre committed:
Merge branch 'develop' into add/3rd-nf-core-deps

marcodelapierre committed:
Merge branch 'add/3rd-nf-core-deps' into add/4th-nf-core-deps

marcodelapierre committed:
Merge branch 'add/4th-nf-core-deps' into update/nextflow-tools

marcodelapierre committed:
Update var/spack/repos/builtin/packages/py-dnspython/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

marcodelapierre committed:
Update var/spack/repos/builtin/packages/py-fastapi-utils/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

marcodelapierre committed:
Update var/spack/repos/builtin/packages/py-fastapi-utils/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

marcodelapierre committed:
Update var/spack/repos/builtin/packages/py-fastapi-utils/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

marcodelapierre committed:
Update var/spack/repos/builtin/packages/py-lagom/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

marcodelapierre committed:
Update var/spack/repos/builtin/packages/py-supervisor/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

marcodelapierre committed:
Merge branch 'add/3rd-nf-core-deps' into add/4th-nf-core-deps

marcodelapierre committed:
Merge branch 'add/4th-nf-core-deps' into update/nextflow-tools

marcodelapierre committed:
Merge branch 'develop' into add/3rd-nf-core-deps

marcodelapierre committed:
Merge branch 'add/3rd-nf-core-deps' into add/4th-nf-core-deps

marcodelapierre committed:
Merge branch 'add/4th-nf-core-deps' into update/nextflow-tools

marcodelapierre committed:
Merge branch 'develop' into add/4th-nf-core-deps

marcodelapierre committed:
Merge branch 'add/4th-nf-core-deps' into update/nextflow-tools

marcodelapierre committed:
Merge branch 'develop' into add/4th-nf-core-deps

marcodelapierre committed:
Merge branch 'add/4th-nf-core-deps' into update/nextflow-tools

marcodelapierre committed:
Update var/spack/repos/builtin/packages/py-social-auth-core/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

marcodelapierre committed:
Merge branch 'develop' into add/4th-nf-core-deps

marcodelapierre committed:
Merge branch 'add/4th-nf-core-deps' into update/nextflow-tools

marcodelapierre committed:
Merge branch 'develop' into add/4th-nf-core-deps

marcodelapierre committed:
fixes from PR review

marcodelapierre committed:
adding missing deps, from PR review

marcodelapierre committed:
py-galaxy2cwl from github tarball, as per PR review

marcodelapierre committed:
Merge branch 'add/4th-nf-core-deps' into update/nextflow-tools

marcodelapierre committed:
fix to py-tuswsgi, as per PR review

marcodelapierre committed:
Merge branch 'add/4th-nf-core-deps' into update/nextflow-tools

marcodelapierre committed:
Merge branch 'develop' into update/nextflow-tools

adamjstewart commented on var/spack/repos/builtin/packages/py-galaxy-tool-util/package.py:
```diff
@@ -0,0 +1,99 @@
+    version("22.1.5", sha256="60e0372f16255c5c11ec5c49dff432ed3beb97123d026f463cf633bc605c0112")
+
+    depends_on("python@3.7:3.10", type=("build", "run"))
+    depends_on("py-poetry", type="build")
```
22.1.5 uses setuptools, not poetry

marcodelapierre replied:
done, thanks

adamjstewart commented on var/spack/repos/builtin/packages/py-galaxy-tool-util/package.py:
```diff
@@ -0,0 +1,99 @@
+    depends_on("py-uvloop", type=("build", "run"))
+    depends_on("py-webob", type=("build", "run"))
+    depends_on("py-whoosh", type=("build", "run"))
+    depends_on("py-zipstream-new", type=("build", "run"))
```
22.1.5 only has the following dependencies:
```
galaxy-util>=22.1
galaxy-containers>=22.1
lxml
pydantic
pyyaml
sortedcontainers
typing-extensions
```

marcodelapierre replied:
done

adamjstewart commented on var/spack/repos/builtin/packages/py-galaxy-util/package.py:
```diff
@@ -0,0 +1,96 @@
+    depends_on("python@3.7:3.10", type=("build", "run"))
+    depends_on("py-poetry", type="build")
+
+    depends_on("py-a2wsgi", type=("build", "run"))
```
Same comments here, 22.1.2 does not restrict Python, uses setuptools instead of poetry, and only has the following deps:
```
bleach
boltons
docutils
importlib_resources
MarkupSafe
packaging<22  # packaging 22.0 dropped LegacyVersion
pycryptodome
pyparsing
PyYAML
requests
Routes
typing-extensions
zipstream-new
```

marcodelapierre replied:
done

adamjstewart commented on var/spack/repos/builtin/packages/py-pulsar-galaxy-lib/package.py:
```diff
@@ -0,0 +1,25 @@
+    depends_on("py-webob", type=("build", "run"))
+    depends_on("py-psutil", type=("build", "run"))
+    depends_on("py-pastedeploy", type=("build", "run"))
+    depends_on("py-pyyaml", type=("build", "run"))
```
Missing deps:
```
galaxy-job-metrics>=19.9.0.dev1
galaxy-objectstore>=19.9.0.dev1
galaxy-tool-util>=19.9.0.dev3
galaxy-util>=19.9.0.dev1
```
Just drop the dev*, Spack sorts dev as newer than not-dev

marcodelapierre replied:
done

marcodelapierre committed:
Merge branch 'develop' into update/nextflow-tools

marcodelapierre committed:
Merge branch 'develop' into update/nextflow-tools

marcodelapierre committed:
nf-tools: edits from PR review

marcodelapierre committed:
adding 3x more galaxy deps

marcodelapierre committed:
fix

adamjstewart commented on var/spack/repos/builtin/packages/py-galaxy-util/package.py:
```diff
@@ -0,0 +1,32 @@
+    depends_on("py-docutils", type=("build", "run"))
+    depends_on("py-importlib-resources", type=("build", "run"))
+    depends_on("py-markupsafe", type=("build", "run"))
+    depends_on("py-packaging@:22", type=("build", "run"))
```
```suggestion
    depends_on("py-packaging@:21", type=("build", "run"))
```

adamjstewart reviewed (changes requested):
One last change and this should be good to go! The PR is still marked as a draft, you'll have to mark it as "ready for review".

marcodelapierre committed:
fixing circular dep of py-poetry-plugin-export with py-poetry

marcodelapierre committed:
added newest nf-core-tools version

marcodelapierre committed:
Update var/spack/repos/builtin/packages/py-galaxy-util/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

adamjstewart commented on var/spack/repos/builtin/packages/py-poetry-plugin-export/package.py:
```diff
@@ -15,5 +15,5 @@ class PyPoetryPluginExport(PythonPackage):
 
     depends_on("python@3.7:3", type=("build", "run"))
-    depends_on("py-poetry@1.2:1", type=("build", "run"))
+    # depends_on("py-poetry@1.2:1", type=("build", "run")) # circular dependency
```
@alalazo FYI another circular dependency, unfortunately at run-time. @pradyunsg how does pip handle this kind of situation?

marcodelapierre replied:
checked in the source code, it is actually a run-time only dependency

adamjstewart replied:
Can we move this to a separate PR so we can discuss this in more detail without holding up this PR? I'm not sure the best way to handle this.

pradyunsg replied:
Circular build dependencies are not permitted w/ pip.

An individual environment graph is permitted to have a cycle in it though, which is guarenteed to be broken at an arbitrary point when installing packages.

marcodelapierre replied:
> Can we move this to a separate PR so we can discuss this in more detail without holding up this PR? I'm not sure the best way to handle this.

Without fixing this, `nf-core-tools` does not even concretise, so on my end I am happy to hold the PR. But happy to do otherwise if you think it is better to completely decouple the two, e.g. for the sake of better managing them.

adamjstewart replied:
Checked the latest versions of both poetry and poetry-plugin-export but the circular dep still exists. Looks like the developers are aware of this: https://github.com/python-poetry/poetry/pull/5980

I'm not really sure what to do in the meantime. I think the fix you've proposed is about the best we can do. The `py-poetry-plugin-export` package will be broken unless users also install `py-poetry`, but `py-poetry` is probably far more popular and more important to get working.

marcodelapierre replied:
Oh, I forgot to mention one aspect, that makes the current workaround practically quite effective.
In Spack, the only dependent of `py-poetry-plugin-export` is actually `py-poetry`. So if the latter is installed the former will be, too, hence it is unlikely to have a situation where the latter is broken.

marcodelapierre committed:
Merge branch 'develop' into update/nextflow-tools

marcodelapierre commented:
@adamjstewart build and small test run both successful.
ready to go!

marcodelapierre committed:
fix in py-poetry-plugin-export

marcodelapierre committed:
Merge branch 'develop' into update/nextflow-tools

marcodelapierre commented:
@spackbot run pipeline

marcodelapierre committed:
Merge branch 'develop' into update/nextflow-tools

marcodelapierre commented:
@spackbot run pipeline

marcodelapierre commented:
@pditommaso @ewels , it took a while to get through 70+ new recipes for some of its dependencies, but it is now here .. the recipe for `nf-core-tools` is available on the `develop` branch of Spack!

marcodelapierre commented:
Thanks again @adamjstewart for your invaluable support and work, I have really appreciated it!

ewels commented:
Hooray! Thanks @marcodelapierre - I was getting notified about progress on this PR, kudos on getting it through in the end!

pditommaso commented:
<img src="https://media2.giphy.com/media/A8UAkFFXglXkEhfoWn/giphy.gif"/>
